src/blenderbim/blenderbim/bim/module/spreadsheet_writer/operator.py

```diff
@@ -0,0 +1,652 @@
+#print ('pandas',pd.__version__, pd.__file__)
+#print ('xlsxwriter',xlsxwriter.__version__, xlsxwriter.__file__)
+
+#https://stackoverflow.com/questions/72657415/fix-futurewarning-related-to-the-pandas-append-function
+
+#function to open file on windows, mac and linux
+def open_file(filename):
+    if sys.platform == "win32":
+        os.startfile(filename)
+    else:
+        opener = "open" if sys.platform == "darwin" else "xdg-open"
+        subprocess.call([opener, filename])
+ 
+ 
+        
+class Element(list):
+    def __init__(self, name, attrs):
+        self.name = name
+        self.attrs = attrs
+
+
+class TreeBuilder:
+    def __init__(self):
+        self.root = Element("root", None)
+        self.path = [self.root]
+    def start_element(self, name, attrs):
+        element = Element(name, attrs)
+        self.path[-1].append(element)
+        self.path.append(element)
+    def end_element(self, name):
+        assert name == self.path[-1].name
+        self.path.pop()
+    def char_data(self, data):
+        self.path[-1].append(data)
+
+
+def get_hidden_rows(node):
+    row = 0
+    for e in node:
+        if not isinstance(e, Element):
+            continue
+        yield from get_hidden_rows(e)
+        if e.name != "table:table-row":
+            continue
+        attrs = e.attrs
+        rows = int(attrs.get("table:number-rows-repeated", 1))
+        if "table:visibility" in attrs.keys():  # If the key is here, we can assume it's hidden (or can we ?)
+            #for row_idx in range(rows):
+            #    yield row + row_idx
+            yield from range(row, row + rows)
+        row += rows
+
+        
+        
+
+class ConstructDataFrame:
+
+    def __init__(self, context):
+        
+        blenderbim_spreadsheet_properties = context.scene.blenderbim_spreadsheet_properties
+        my_collection = context.scene.my_collection
+        
+        ifc_dictionary = defaultdict(list)
+        ifc_custom_property_list = []
+        
+        blenderbim_spreadsheet_properties.my_workbook = 'Overview'
+        blenderbim_spreadsheet_properties.my_xlsx_file  = IfcStore.path.replace('.ifc','_blenderbim.xlsx') 
+        blenderbim_spreadsheet_properties.my_ods_file = IfcStore.path.replace('.ifc','_blenderbim.ods') 
+
+        
+        ifc_file = ifcopenshell.open(IfcStore.path)
+        products = ifc_file.by_type('IfcProduct')
+        
+        ifc_version = (ifc_file.schema)
+        print ("IFC version found: ", ifc_version)
+        
+        is_external = 'IsExternal'
+        loadbearing = 'LoadBearing'
+        fire_rating = 'FireRating'
+        
+        length = 'Length'
+        width = 'Width'
+        height = 'Height'
+        area = 'Area' 
+        netarea = 'NetArea'
+        netsidearea = 'NetSideArea'
+        volume = 'Volume'
+        netvolume = 'NetVolume'
+        perimeter = 'Perimeter'
+
+        for product in products:
+            ##################################################################
```

Comment:
Instead of using comments as delimiters it is better practice to decompose bigger functions into multiple smaller ones. A good length to aim for in a function is 7 - 10 lines top.